machine/v1beta1/types_machine.go

```diff
@@ -194,6 +199,38 @@ type MachineSpec struct {
 	ProviderID *string `json:"providerID,omitempty"`
 }
 
+// LifecycleHooks allow users to pause operations on the machine at
+// certain prefedined points within the machine lifecycle.
+type LifecycleHooks struct {
+	// PreDrain hooks prevent the machine from being drained.
+	// This also blocks further lifecycle events, such as termination.
+	// +optional
+	PreDrain []LifecycleHook `json:"preDrain,omitempty"`
+
+	// PreTerminate hooks prevent the machine from being terminated.
+	// PreTerminate hooks be actioned after the Machine has been drained.
+	// +optional
+	PreTerminate []LifecycleHook `json:"preTerminate,omitempty"`
+}
+
+// LifecycleHook represents a single instance of a lifecycle hook
+type LifecycleHook struct {
+	// Name defines a unique name for the lifcycle hook.
+	// The name should be unique and descriptive, ideally 1-3 words, in camelCase with no spaces.
```

Comment:
is "camelCase" specified by the upstream?  If not, we normally use CamelCase.

Reply:
That would be a typo, I had intended (from the examples I've created) that this be `CamelCase`, good catch, will update